include/amqpcpp/login.h

```diff
@@ -39,28 +39,79 @@ class Login
      */
     std::string _password;
 
+    /**
+     *  The security mechanism
+     *  @var string
+     */
+    std::string _mechanism;
+
+    /**
+     *  The security response
+     *  @var string
+     */
+    std::string _response;
+
 
 public:
     /**
-     *  Default constructor
+     *  Default constructor with PLAIN authentication
      */
-    Login() : _user("guest"), _password("guest") {}
+    Login() : _user("guest"), _password("guest")
+    {
+        // Set default PLAIN mechanism and response
+        _mechanism = "PLAIN";
+        _response = saslPlain();
+    }
 
     /**
-     *  Constructor
+     *  Constructor with PLAIN authentication mechanism
+     *
      *  @param  user
      *  @param  password
      */
     Login(std::string user, std::string password) :
-        _user(std::move(user)), _password(std::move(password)) {}
+        _user(std::move(user)), _password(std::move(password))
+    {
+        // Set default PLAIN mechanism and response
+        _mechanism = "PLAIN";
+        _response = saslPlain();
+    }
 
     /**
-     *  Constructor
+     *  Constructor with PLAIN authentication mechanism
      *  @param  user
      *  @param  password
      */
     Login(const char *user, const char *password) :
-        _user(user), _password(password) {}
+        _user(user), _password(password)
+    {
+        // Set default PLAIN mechanism and response
+        _mechanism = "PLAIN";
+        _response = saslPlain();
+    }
+
+    /**
+     *  Constructor with custom authentication mechanism
+     *  @param  user
+     *  @param  password
+     *  @param  mechanism
+     *  @param  response
+     */
+    Login(std::string user, std::string password, std::string mechanism, std::string response)
+        : _user(std::move(user))
+        , _password(std::move(password))
+        , _mechanism(std::move(mechanism))
+        , _response(std::move(response)) {}
+
+    /**
+     *  Constructor with custom authentication mechanism
+     *  @param  user
+     *  @param  password
+     *  @param  mechanism
+     *  @param  response
+     */
+    Login(const char *user, const char *password, const char *mechanism, const char *response)
+        : _user(user), _password(password), _mechanism(mechanism), _response(response) {}
```

Comment:
Can you add more in-code documentation? I don't understand the use case where it is up to the caller to supply a username, password, mechanism and response. What sort of values is one supposed to pass here?
```c++
Login mylogin("my-username", "my-password", "my-special-login-mechanism", "bla bla bla");
```
Is this permitted?